src/libopensc/cards.h

```diff
@@ -138,6 +138,7 @@ enum {
 	SC_CARD_TYPE_PIV_II_OBERTHUR,
 	SC_CARD_TYPE_PIV_II_PIVKEY,
 	SC_CARD_TYPE_PIV_II_SWISSBIT,
+	SC_CARD_TYPE_PIV_II_800_73_4,
```

Comment:
`SC_CARD_TYPE_PIV_II_800_73_4` right now defines any PIV card that indicates it support SM or other 800-73-4 feature. These are in the response to SELECT_FILE for the PIV AID and the "0x7E interindustry tag (Discovery Object)". If there are other cards or "PIV like applets" that do not follow the standard they can be addressed in future commits.
The features are contained in other fields in `piv_private_data_t` as  `pin_policy`, `sm_flags` and `cs`. 
  
`SC_CARD_TYPE_PIV_II_GENERIC` is similar. If card responds to SELECT_FILE for the PIV AID or GET DATA for "0x7E interindustry tag (Discovery Object)"  and response contains the PIV_AID it is considered as `SC_CARD_TYPE_PIV_II_GENERIC`. The match routines will also look at the ATR and can spot Yubico cards. That is why the match routines are so complicated. Thy try to test a card without resetting the login state. With PIV_SM,  sending a SM  protected `7E` can also test if SM is still intact.    

If there are other cards or "PIV like applets" that do not follow the standard they can be addressed in future commits. Any one can write an open source "PIV like applet" to any token or card and the card-piv.c will try and use it.  This has been the practice from day one. As some "PIV like applet" is found to have some problems it can be assigned a card type, and  "CI_" flags  can be set that define what is(are) the problem(s). There are already 17 card issue flags so far.  See https://github.com/OpenSC/OpenSC/blob/master/src/libopensc/card-piv.c#L310-L325

Reply:
OK, we'll handle new issues when they appear.